### PR TITLE
Upload logs to Amazon S3

### PR DESCRIPTION
**Pull request by simong**

### CODE REVIEW

**Comment by simong** [comment]
Confirmed to be working

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,11 @@ branches:
   only:
     - master
 
+env:
+  global:
+    - secure: "YRS6gw2FhSlkRgXmzvyAdqEGLVZozrS4pESBlbX8Zi9zVIAI2+ExJb/GlhWSvcePeRBegvqE+/9K/joDEecEUacf0Ky7StCNZ4l4sKnE65ut7XOXwbhwKfOOHuFIqvrLMRKN+xFy3JrvN22v2dAoP5lR5yyz6pf6rX0PJjx0gYs="
+    - secure: "E0yI8WcCo5YdHywbMgyRRE3a/WbzXMnvCZsjPP0cPjFDjzqqDLefUoNKKxgDJDPTcreM7DjkXauIt6tw85hEMlvsWJ0svVWLFX9tNIt0EG3Tv3MxEoKkWfKBHC7JYpcwLVM9HoYVfdVBfdu0RAefrgdM+28HwCG/Q+toSPAuzdg="
```
Comment (nicolaasmatthijs) [inline .travis.yml]: Do we need 2 secures in here?

File: etc/scripts/travis/upload_logs [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
+
+var fs = require('fs');
+var amazons3 = require('awssum-amazon-s3');
```
Comment (nicolaasmatthijs) [inline etc/scripts/travis/upload_logs]: can you do the usual alphabetic here?

File: etc/scripts/travis/upload_logs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    .argv;
+
+// Display the help if requested
+if (argv.h) {
+    require('yargs').showHelp();
+    return process.exit(0);
+}
+
+// Ensure that we're on the master branch
```
Comment (nicolaasmatthijs) [inline etc/scripts/travis/upload_logs]: I guess technically it ensure that we're looking at a PR to the master branch?

File: etc/scripts/travis/upload_logs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+// Get the argument values
+var bucketName = argv.b;
+var region = argv.r;
+var logPath = argv.l;
+
+// Stat the log file to a/ ensure that it exists and b/ get the size of the file
+fs.stat(logPath, function(err, stat) {
+    if (err) {
+        console.log('Failed to get the file information for the test log file');
```
Comment (nicolaasmatthijs) [inline etc/scripts/travis/upload_logs]: You can merge the line below into this one

File: etc/scripts/travis/upload_logs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        'region': region
+    });
+    s3.PutObject({
+        'BucketName': bucketName,
+        'ObjectName': objectName,
+        'ContentLength': stat.size,
+        'Body': fs.createReadStream(logPath)
+    }, function(err) {
+        if (err) {
+            console.log('Failed to upload to S3');
```
Comment (nicolaasmatthijs) [inline etc/scripts/travis/upload_logs]: Same here

**Comment by nicolaasmatthijs** [comment]
Left a few comments. Back to @simong for follow-up

File: .travis.yml (hunk 1 above)
Comment (simong) [inline .travis.yml]: Yes, one for the access key and another for the secret key

File: etc/scripts/travis/upload_logs (hunk 4 above)
Comment (simong) [inline etc/scripts/travis/upload_logs]: Not really, that would stringify the `err` object, it's a bit easier to debug if the full error object is logged.

**Comment by simong** [comment]
Pushed some follow-up

**Comment by nicolaasmatthijs** [comment]
Looks good. Merged

**Pull request merged**